fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/node/PredicateParser.java

```diff
@@ -0,0 +1,421 @@
+    /**
+     * Add a NOT node to the expression
+     */
+    public void not() {
+        addNode(new NotExpNode());
+    }
+
+    /**
+     * Add a BETWEEN node to the expression
+     */
+    public void between() {
+        addNode(new BetweenExpNode());
+    }
+
+    /**
+     * Add an addition + node to the expression
+     */
+    public void add() {
+        addNode(new AddExpNode());
+    }
+
+    /**
+     * Add a subtraction node to the expression
+     */
+    public void sub() {
+        addNode(new SubExpNode());
+    }
+
+    /**
+     * Add a multiplication node to the expression
+     */
+    public void mult() {
+        addNode(new MultExpNode());
+    }
+
+    /**
+     * Add a division node to the expression
+     */
+    public void div() {
+        addNode(new DivExpNode());
+    }
+
+    /**
+     * Add a LEFT PAREN ( node to the expression
+     */
+    public void leftParen() {
+        addNode(new LeftParenExpNode());
+    }
+
+    /**
+     * Add a RIGHT PAREN ) node to the expression
+     */
+    public void rightParen() {
+        addNode(new RightParenExpNode());
+    }
+
+    /**
+     * Add a bind marker ? and its value to the expression
+     * @param node
+     */
+    public void bindMarker(BindMarkerNode node) {
+        addNode(node);
+    }
+
+    /**
+     * Convenience function to add an IN list of literal string values. This should be used
+     * only where the string values are internal values, never input data. If the values
+     * come from input data, then for security it is important to use bind markers instead.
+     * @param inList
+     */
+    public void inLiteral(String[] inList) {
+        // Parameters are encoded for security, but bind markers are always preferred and
+        // recommended for security.
+        List<ExpNode> args = Arrays.asList(inList).stream().map(v -> new StringExpNode(encodeParameter(v))).collect(Collectors.toList());
+        addNode(new InListExpNode(args));
+    }
+
+    /**
+     * Convenience function to add an IN list of literal long values.
+     * @param inList
+     */
+    public void inLiteral(Long[] inList) {
+        List<ExpNode> args = Arrays.asList(inList).stream().map(v -> new LongExpNode(v)).collect(Collectors.toList());
+        addNode(new InListExpNode(args));
+    }
+
+    /**
+     * Convenience function to add an IN list of `<bindMarkerCount>` bind markers
+     * @param bindMarkerCount
+     */
+    public void inString(List<String> inList) {
+        List<ExpNode> args = inList.stream().map(v -> new StringBindMarkerNode(v)).collect(Collectors.toList());
+        addNode(new InListExpNode(args));
+    }
+
+    /**
+     * Convenience function to add an IN list of `<bindMarkerCount>` bind markers
+     * @param bindMarkerCount
+     */
+    public void inLong(List<Long> inList) {
+        List<ExpNode> args = inList.stream().map(v -> new LongBindMarkerNode(v)).collect(Collectors.toList());
+        addNode(new InListExpNode(args));
+    }
```

Comment:
I didn't seem the stack trace where these are referenced.

Reply:
It's a framework. These are available for future use.